### PR TITLE
idp/adfs: Add an ADFS identity provider

### PR DESCRIPTION
**Pull request by mhilton**
Create an identity provider that authenticates with an Active Directory
Federation Services instance using OpenID Connect.

Note that QA for this identity provider used Azure AD as a substitute
for ADFS.

### CODE REVIEW

File: idp/adfs/adfs.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	"github.com/canonical/candid/idp/openid"
+)
+
+func init() {
+	idp.Register("adfs", func(unmarshal func(interface{}) error) (idp.IdentityProvider, error) {
+		var p Params
+		if err := unmarshal(&p); err != nil {
+			return nil, errgo.Notef(err, "cannot unmarshal adfs parameters")
+		}
+		if p.URL == "" {
```
Comment (alesstimec) [inline idp/adfs/adfs.go:22]: nitpick: i'd put this checks in 

func (p *Params) validate() error {
}

**Review by alesstimec — APPROVED** [review]
LGTM

File: idp/adfs/adfs.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	// the application as generated on
+	// https://console.developers.google.com/apis/credentials.
+	ClientSecret string `yaml:"client-secret"`
+
+	// Hidden is set if the IDP should be hidden from interactive
+	// prompts.
+	Hidden bool `yaml:"hidden"`
+}
+
+// NewIdentityProvider creates a google identity provider with the
```
Comment (TeodorPt) [inline idp/adfs/adfs.go]: nit: ADFS instead of google?

File: idp/adfs/adfs.go (hunk 1 above)
Comment (mhilton) [reply idp/adfs/adfs.go:22]: I'm going to keep[ this as is as we do this in a few places, I might change this a follow-up in a number of IDPs though.

File: idp/adfs/adfs.go (hunk 2 above)
Comment (mhilton) [reply idp/adfs/adfs.go]: Thanks, I found a couple of others I've fixed too.

**Pull request merged**